cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSQLEngineAdapter.java

```diff
@@ -807,4 +812,27 @@ protected static void countStageMetrics(StageMetrics stageMetrics,
       return writeResult.isSuccessful();
     });
   }
+
+  /**
+   * Thread factory for SQL Engine Tasks. This thread factory ensures the parent context's classloader gets passed
+   * into the child threads.
+   */
+  private class SQLEngineAdapterThreadFactory implements ThreadFactory {
+    private final AtomicLong threadCounter;
+    private final ClassLoader classLoader;
+    private static final String NAME_TEMPLATE = "batch-sql-engine-adapter-%d";
+
+    SQLEngineAdapterThreadFactory(ClassLoader classLoader) {
+      this.threadCounter = new AtomicLong(0L);
+      this.classLoader = classLoader;
+    }
+
+    @Override
+    public Thread newThread(Runnable r) {
+      Thread t = new Thread(r);
+      t.setName(String.format(NAME_TEMPLATE, threadCounter.getAndIncrement()));
+      t.setContextClassLoader(classLoader);
```

Comment:
Please ensure you set daemon flag on the thread

Reply:
Since these are BQ Pushdown execution threads, I was thinking it makes sense to have them as high priority/realtime threads. WDYT?

Reply:
discussed offline. Threads set as daemon.